core/jazz_slack-channel/index.js

```diff
@@ -0,0 +1,504 @@
+const responseObj = require("./components/response.js"); //Import the response module.
+const configObj = require("./components/config.js"); //Import the environment data.
+const logger = require("./components/logger.js")();
+const request = require("request");
+
+module.exports.handler = (event, context, cb) => {
+
+  //Initializations
+  let errorHandler = errorHandlerModule(),
+  config = configObj.getConfig(event, context);
+  logger.init(event, context);
+  let isServAccRequested = false;
+
+  genericInputValidation(event, config)
+    .then(() => createPublicSlackChannel(config, event.body))
+    .then(res => addMemberToChannel(config, res.data, isServAccRequested, event.body))
+    .then(res => {
+      logger.info("successfully created public slack channel: " + JSON.stringify(res));
+      let result = {
+        "channel_id": res.data.channelInfo.id,
+        "channel_name": res.data.channelInfo.name,
+        "channel_link": res.data.channelInfo.link,
+        "members": res.data.slackChannelMembers,
+      };
+      return cb(null, responseObj(result, event.body));
+    })
+    .catch(error => {
+      logger.error(error);
+      if (error.result === "inputError") {
+        return cb(JSON.stringify(errorHandler.throwInputValidationError(error.message)));
+      } else if (error.result === "unauthorized") {
+        return cb(JSON.stringify(errorHandler.throwUnauthorizedError(error.message)));
+      } else {
+        return cb(JSON.stringify(errorHandler.throwInternalServerError(error.message)));
+      }
+    });
+};
+
+function genericInputValidation(event, config) {
+  logger.debug("Inside genericInputValidation");
+  return new Promise((resolve, reject) => {
+    if (!event || !event.body || !event.method) {
+      reject({
+        result: "inputError",
+        message: "Service inputs not defined!"
+      });
+    }
+
+    if (!event.body.channel_name) {
+      reject({
+        result: "inputError",
+        message: "channel_name is required!."
+      });
+    } else {
+      let name = event.body.channel_name;
+      if (name.length > config.slack_channel_name_length) {
+        reject({
+          result: "inputError",
+          message: "channel name should not exceed more than 21 characters"
+        });
+      } else if (/[^a-zA-Z0-9\-\_]/.test(name)) {
+        reject({
+          result: "inputError",
+          message: "channel name should accept letters, numbers, hyphens, and underscores"
+        });
+      }
+    }
+
+    if (!event.body.users || event.body.users.length < 1) {
+      reject({
+        result: "inputError",
+        message: "users.email_id is required!."
+      });
+    } else {
+      let users = event.body.users;
+      for (let i in users) {
+        let emailAddress = users[i].email_id;
+        if (!validateEmailId(emailAddress)) {
+          let msg = 'Not a valid email id!';
+          if (emailAddress) {
+            msg = emailAddress + " not a valid email id!.";
```

Comment:
Update message to `emailAddress + " is not a valid email address."`

Reply:
Updated